From d918303a1cf2ea33e5b67e96b671112d7a1df78e Mon Sep 17 00:00:00 2001
From: Simo Sorce <simo@redhat.com>
Date: Thu, 31 Jul 2014 13:36:03 -0400
Subject: [PATCH 1/3] Missing variable initialization

If name_id s unconditonally dereferenced in the cleanup code.
If it is not initialized it may cause segfaults or other misbehaviors.

License: MIT
Signed-off-by: Simo Sorce <simo@redhat.com>
---
 lasso/id-ff/session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lasso/id-ff/session.c b/lasso/id-ff/session.c
index 4cb3e2b8b530abf8453f5729062e77e3883478d8..9446e2855536d020f3ca9636901d79ca7c7df824 100644
--- a/lasso/id-ff/session.c
+++ b/lasso/id-ff/session.c
@@ -858,7 +858,7 @@ init_from_xml_nid_and_session_index(LassoNode *node, xmlNode *nid_and_session_in
 	xmlChar *provider_id = NULL;
 	xmlChar *assertion_id = NULL;
 	xmlNode *nid;
-	LassoNode *name_id;
+	LassoNode *name_id = NULL;
 	struct _NidAndSessionIndex *nid_and_session_index;
 
 	provider_id = xmlGetProp(nid_and_session_index_node, BAD_CAST PROVIDER_ID);
-- 
1.9.3

